x/interchainqueries/types/params.go

```diff
@@ -1,30 +1,49 @@
 package types
 
 import (
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
 	"gopkg.in/yaml.v2"
 )
 
 var _ paramtypes.ParamSet = (*Params)(nil)
 
+var (
+	KeyQuerySubmitTimeout               = []byte("QuerySubmitTimeout")
+	DefaultQuerySubmitTimeout           = uint64(518400) // One month, with block_time = 5s
+	KeyQueryDeposit                     = []byte("QueryDeposit")
+	DefaultQueryDeposit       sdk.Coins = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(int64(1_000_000))))
+)
+
 // ParamKeyTable the param key table for launch module
 func ParamKeyTable() paramtypes.KeyTable {
-	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+	return paramtypes.NewKeyTable(
+		paramtypes.NewParamSetPair(KeyQuerySubmitTimeout, DefaultQuerySubmitTimeout, func(value interface{}) error { return nil }),
+		paramtypes.NewParamSetPair(KeyQueryDeposit, sdk.Coins{}, validateCoins),
+	)
 }
 
 // NewParams creates a new Params instance
-func NewParams() Params {
-	return Params{}
+func NewParams(querySubmitTimeout uint64, queryDeposit sdk.Coins) Params {
+	return Params{
+		QuerySubmitTimeout: querySubmitTimeout,
+		QueryDeposit:       queryDeposit,
+	}
 }
 
 // DefaultParams returns a default set of parameters
 func DefaultParams() Params {
-	return NewParams()
+	return NewParams(DefaultQuerySubmitTimeout, DefaultQueryDeposit)
 }
 
 // ParamSetPairs get the params.ParamSet
 func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
-	return paramtypes.ParamSetPairs{}
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyQuerySubmitTimeout, &p.QuerySubmitTimeout, func(value interface{}) error { return nil }),
+		paramtypes.NewParamSetPair(KeyQueryDeposit, &p.QueryDeposit, validateCoins),
```

Comment:
is there any usage of these validate functions at all, if below in the `(p Params) Validate()` we simply return nil?

Reply:
It is required by `NewParamSetPair`